src/mpid/ch4/src/ch4_wait.h

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#ifndef CH4_WAIT_H_INCLUDED
+#define CH4_WAIT_H_INCLUDED
+
```

Comment:
Maybe a better file name since it includes `Test` APIs too? How about `ch4_progress_funcs.h`?

Reply:
I think `ch4_progress_funcs.h` is more ambiguous. I think this is not a big deal, for example, we are doing this quite often, for example, having `finalize` inside `xxx_init.c`.